contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpPushDownListener.java

```diff
@@ -82,9 +85,20 @@ public ScanPushDownListener builderFor(GroupScan groupScan) {
 
     HttpScanPushDownListener(HttpGroupScan groupScan) {
       this.groupScan = groupScan;
+      // Add fields from config
       for (String field : groupScan.getHttpConfig().params()) {
         filterParams.put(field, field);
       }
+
+      // Add fields from the URL path as denoted by {}
+      HttpUrl url = HttpUrl.parse(groupScan.getHttpConfig().url());
+      if (url != null) {
+        List<String> urlParams = SimpleHttp.getURLParameters(url);
+        assert urlParams != null;
```

Comment:
Is there a old code block? the assert will throw `AssertionError` when false.

Reply:
Fixed.  I actually changed the `getUrlParameters()` method such that if it encounters any error, it returns an empty list rather than `null`.  My sense is that is better design.